release-management.md

```diff
@@ -0,0 +1,148 @@
+# go-cose Release Management
+
+## Overview
+
+This document describes [go-cose][go-cose] project release management, which includes release criteria, versioning, supported releases, and supported upgrades.
+
+The go-cose project maintainers strive to provide a stable go-lang implementation for interacting with [COSE][ietf-cose] constructs.
+Stable implies appropriate and measured changes to the library assuring consumers have the necessary functionality to interact with COSE objects.
+If you or your project require added functionality, or bug fixes, please open an issue or create a pull request.
+The project welcomes all contributions from adding functionality, implementing testing to security reviews.
+Please see [here](https://github.com/veraison#contributing) for how to contribute.
+
+## Release Versioning
+
+Consumers of the go-cose project may build directly from main, or pull from released builds.
+Builds from main must reference the git-commit as the version: `v1.0.0-2300d5c`
+
+All go-cose [releases][releases] follow a go-lang flavored derivation (`v*`) of the [semver][sem-ver] format, with optional pre-release labels.
+
+Logical Progression of a release: `v1.0.0-2300d5c` --> `v1.0.0-alpha.1` --> `v1.0.0-alpha.2` --> `v1.0.0-rc.1` --> `v1.0.0`
+
+A new major or minor release will not have an automated build/release posted until the branch reaches alpha quality.
+
+- all versions use a preface of `v`
+- `X` is the [Major](#major-releases) version
+- `Y` is the [Minor](#minor-releases) version
+- `Z` is the [Patch](#patch-releases) version
+- _Optional_ `-alpha.n` | `-rc.n` [pre-release](#pre-release) version
+  - Each incremental alpha or rc build will bump the suffix (`n`) number.
+  - It's not expected to have more than 9 alphas or rcs.
+The suffix will be a single digit.
+  - If > 9 builds do occur, the format will simply use two digit indicators (`v1.0.0-alpha.10`)
+
+_**Note**: Pre-releases will NOT use the github pre-release flag._
+
+## Branch Management
+
+To meet the projects stability goals, go-cose does not currently operate with multiple feature branches.
+All active development happens in main.
+For each release, a branch is created for servicing, following the versioning name.
+`v1.0.0-alpha-1` would have an associated [v1.0.0-alpha.1](https://github.com/veraison/go-cose/tree/v1.0.0-alpha.1) branch.
+All version and branch names are lower case.
+
```

Comment:
Do we want to maintain all the Release Branches..?

Reply:
I believe the answer should be yes. Even if something is unsupported, or even has a bug/issue, a consumer may be dependent upon it until they can make changes, or take time to take a newer version.